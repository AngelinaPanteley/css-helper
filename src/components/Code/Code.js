import React, { PureComponent } from 'react';
import styles from './Code.scss';
import Pattern from './Pattern/Pattern';

class Code extends PureComponent {
  render() {
    let styleString = '';
    for (let styleKey in this.props.styles) {
      styleString = `${styleString}\n  ${styleKey}: ${this.props.styles[styleKey]};`;
    }
    const styleCode = `.${this.props.previewClass} {${styleString}\n}`;
    return (
      <div className={styles.CodeWrapper}>
        <Pattern code={styleCode}
          title='Styles'
          onCopy={this.props.onCopy}
        />
        <Pattern code={this.props.template.replace(/\n/, "\n\u00a0\u00a0")}
          title='Markup'
          onCopy={this.props.onCopy}
        />
      </div>
    )
  }
}

export default Code;