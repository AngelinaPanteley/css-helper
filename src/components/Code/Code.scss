.CodeWrapper {
  padding: 30px 20px;
}