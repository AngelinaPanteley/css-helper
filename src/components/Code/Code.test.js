"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';

import Code from './Code.js';

configure({ adapter: new Adapter() });

describe('>>>Code', () => {

  //SNAPSHOTS

  const props = {
    styles: {
      border: '1px solid blue'
    },
    previewClass: 'previewClass',
    onCopy: () => { },
    template: '<div class="previewClass"></div>',
  };

  const codeComponent = <Code {...props} />;

  const component = renderer.create(codeComponent);

  it('renders code correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(codeComponent);
  });

  it('renders two patterns', () => {
    chaiExpect(wrapper.find('Pattern')).to.have.lengthOf(2);
  });
})