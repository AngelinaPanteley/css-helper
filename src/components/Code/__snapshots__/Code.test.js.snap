// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Code renders code correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <p
      className={undefined}
    >
      <span>
        Styles
      </span>
      <button
        className={undefined}
        onClick={[Function]}
      >
        <i
          className="undefined undefined"
        />
      </button>
    </p>
    <pre
      className={undefined}
    >
      <code
        className={undefined}
      >
        .previewClass {
          border: 1px solid blue;
        }
      </code>
    </pre>
  </div>
  <div
    className={undefined}
  >
    <p
      className={undefined}
    >
      <span>
        Markup
      </span>
      <button
        className={undefined}
        onClick={[Function]}
      >
        <i
          className="undefined undefined"
        />
      </button>
    </p>
    <pre
      className={undefined}
    >
      <code
        className={undefined}
      >
        &lt;div class="previewClass"&gt;&lt;/div&gt;
      </code>
    </pre>
  </div>
</div>
`;
